src/Columns/ColumnTuple.cpp

```diff
@@ -42,40 +41,48 @@ std::string ColumnTuple::getName() const
     return res.str();
 }
 
-ColumnTuple::ColumnTuple(MutableColumns && mutable_columns)
+ColumnTuple::ColumnTuple(TupleColumns && columns_)
 {
-    columns.reserve(mutable_columns.size());
-    for (auto & column : mutable_columns)
+    /// A tuple can't be empty (see ColumnTuple::size() for example).
+    if (columns_.empty())
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Tuple cannot be empty");
```

Comment:
This check makes Iceberg tests fail:
https://s3.amazonaws.com/clickhouse-test-reports/59845/7d27c067f53a6a14bb862ad47e8f08345f402e3c/integration_tests__release__[4_4].html

The failed tests are passing in the master despite they're creating wrong ColumnTuple objects.
Before this PR ColumnTuple could be created without nested columns but such ColumnTuples didn't work. Even [ColumnTuple::size()](https://github.com/ClickHouse/ClickHouse/blob/09d3b3c0d04aa2ee0c5661a70bb4c67bfbb0765d/src/Columns/ColumnTuple.h#L51) cannot work without nested columns.

In this PR I've added more checks so now the Iceberg tests are failing.